mlfinlab/tests/test_sb_bagging.py

```diff
@@ -0,0 +1,310 @@
+        cusum_events = cusum_filter(self.data['close'], threshold=0.001)
+        vertical_barriers = add_vertical_barrier(t_events=cusum_events, close=self.data['close'],
+                                                 num_days=2)
+        self.meta_labeled_events = get_events(close=self.data['close'],
+                                              t_events=cusum_events,
+                                              pt_sl=[4, 4],
+                                              target=daily_vol,
+                                              min_ret=0.005,
+                                              num_threads=3,
+                                              vertical_barrier_times=vertical_barriers,
+                                              side_prediction=self.data['side'])
+
+        self.meta_labeled_events.dropna(inplace=True)
+        labels = get_bins(self.meta_labeled_events, self.data['close'])
+
+        # Feature generation
+        features = []
+        X = self.data.copy()
+        X['log_ret'] = X.close.apply(np.log).diff()
+        for win in [2, 5, 10, 20, 25]:
+            X['momentum_{}'.format(win)] = X.close / X.close.rolling(window=win).mean() - 1
+            X['std_{}'.format(win)] = X.log_ret.rolling(window=win).std()
+            X['pct_change_{}'.format(win)] = X.close.pct_change(win)
+            X['diff_{}'.format(win)] = X.close.diff(win)
+
+            for f in ['momentum', 'std', 'pct_change', 'diff']:
+                features.append('{}_{}'.format(f, win))
+
+        # Train/test generation
+        X.dropna(inplace=True)
+        X = X.loc[self.meta_labeled_events.index, :]  # Take only filtered events
+        labels = labels.loc[X.index, :]  # Sync X and y
+        self.meta_labeled_events = self.meta_labeled_events.loc[X.index, :]  # Sync X and meta_labeled_events
+
+        self.X_train, self.y_train_clf, self.y_train_reg = X.iloc[:300][features], labels.iloc[:300].bin, \
+                                                           labels.iloc[:300].ret
+        self.X_test, self.y_test_clf, self.y_test_reg = X.iloc[300:][features], labels.iloc[300:].bin, \
+                                                        labels.iloc[300:].ret
+
+    def test_other_sb_features(self):
+        """
+        This function simply creates various estimators in order to get all lines of code covered, we don't check
+        anything here
+        """
+        clf_1 = RandomForestClassifier(n_estimators=1, criterion='entropy', bootstrap=True,
+                                       class_weight='balanced_subsample', max_depth=12)
+        clf_2 = RandomForestClassifier(n_estimators=1, criterion='entropy', bootstrap=False,
+                                       class_weight='balanced_subsample', max_depth=12)
+        clf_3 = KNeighborsClassifier()
+        clf_4 = LinearSVC()
+
+        sb_clf_1 = SequentiallyBootstrappedBaggingClassifier(base_estimator=clf_1, max_features=0.2,
```

Comment:
Game changer! Looking forward to using this ensemble!

Reply:
Yeah!